db_utils.py

```diff
@@ -38,6 +38,7 @@ def reset_connection(self):
                 self.cnx.close()
             except Exception:
                 pass
+        self.reset = True
```

Comment:
just make self.cnx = None here, that will make self.connect rebuild the connector.